### DISCUSSION TITLE
Error initializing MONAILabel server

### DISCUSSION THREAD

**Discussion by nmahalingam4** [Project-MONAI/MONAILabel#1583] [Q&A]
Hello,

 I am currently running Python 3.8 and have CUDA enabled with MONAIlabel version 0.8.0. I am trying to launch the MONAILabel server that I'm running locally on my computer and I am getting this error when I try to start it:
![image](https://github.com/Project-MONAI/MONAILabel/assets/143131321/df82f8d0-ed89-45e4-801c-659b7eef9951)

I found that there are two copies of the libiomp5md.dll file in my anaconda virtual environment libraries, but when I remove one, my CUDA becomes disabled (torch.cuda.is_available = False). I'm not sure how to resolve this and was wondering if I could get some guidance.

Thank you.

**Comment by diazandr3s** [comment]
Hi @nmahalingam4,

Thanks for opening this discussion.
I'd suggest either starting a clean environment or using the Docker container: https://github.com/project-monai/monailabel#development-version
What do you see after running `nvidia-smi` command in the terminal? Have you tried starting the MONAI Label server and running inference?

Let us know,

**Reply by nmahalingam4** [reply]
Hi @diazandr3s 
When I run nvidia-smi command, I get this output:
![image](https://github.com/Project-MONAI/MONAILabel/assets/143131321/4ce82545-cc87-4045-8e1f-cb33c1315d7a)

I have created a new virtual environment for this particular project that only has one version of CUDA and MONAILabel installed. I have tried to start the MONAI Label server, but I get the previously cited error message:
![image](https://github.com/Project-MONAI/MONAILabel/assets/143131321/c35dc6e5-2878-4aef-93ed-ccd665d32925)

Any guidance is much appreciated!

**Reply by diazandr3s** [reply]
Hi @nmahalingam4,

This doesn't seem a MONAI Label issue. 
I quickly searched on Google for the issue, and they recommend exporting an env variable: https://github.com/explosion/spaCy/issues/8366
Have you tried using the Docker container?

Let us know

**Reply by nmahalingam4** [reply]
Hi @diazandr3s 

Thank you for the suggestion. It seems that it was related to the Anaconda virtual environment that I was using. I used python venv instead and was able to initiate the server and train my model.

I am running the MONAI Label server on my machine, and would like my coworker to run a local 3D Slicer instance on his machine with the MONAI Label server. I am not sure which MONAI Label server IP to give him to be able to connect. I tried to create an ssh tunnel but still get this error:
![image](https://github.com/Project-MONAI/MONAILabel/assets/143131321/93580cc3-d05c-43d3-b837-0d7322ed4099)

I understand that I could use OHIF, but would like to try getting the 3D Slicer instance running as there are more editing tools that would be useful for our use case.

Any guidance is appreciated. Thank you!

**Reply by diazandr3s** [reply]
Hi @nmahalingam4,

Thanks for the update.
You should give your coworker the public IP address of the instance/computer that has MONAI Label running. Also, you have to make sure the port 8000 is open so they can connect to it.

From the print screen you shared, I see you didn't provide the public IP address. you're still using the localhost.

Hope this helps,

**Comment by natachavalador** [comment]
Hi,
It seems I am facing the same problem with the server.

. Running on my own pc (Windows 11)
. 3D Slicer installed with MonaiLabel plug-in working
. Tryed Anaconda, Python 3.11 and Visual Studio Code as an interface
. Followed precisely all tutorials from Andres Pinto and the server remains not working
. Tried to mix my personal IP with the Port :8000 and the error remains

Could you please help me?

KR,
Natacha Valador

**Reply by diazandr3s** [reply]
Hi @natachavalador, 

Thanks for reporting this.

Could you please share the error/logs you're getting when [starting the MONAI Label server](https://github.com/Project-MONAI/MONAILabel?tab=readme-ov-file#getting-started-with-monai-label)? Have you tried using Docker container as well?

**Reply by natachavalador** [reply]
![MonaiLabelError](https://github.com/user-attachments/assets/68641687-f419-48a6-b89b-121b5fe3357f)
Hi,
Thanks for you help.
Attached the error.
Hope it wil be easy to solve :)
KR
